### DISCUSSION TITLE
Pengurusan Anak Kariah (Dashboard + Public)

### DISCUSSION THREAD

**Discussion by rohaizanr** [Dev4w4n/e-masjid.my#147] [Ideas]
Request dari Setiausaha Masjid Sungai Kecil

Sediakan module Pendaftaran Anak Kariah (public)

Contoh workflow/screen:

Google Form (Self initiative dari Setiausaha Masjid Sg Kecil):
https://forms.gle/i1tXJ9PQuFxi2KKV7

Masjid Pro:
https://www.masjidpro.com/Masjid/daftar_online/pendaftaran.php?redirect=&id_masjid=3744

Isu:
Akta 709(Akta Perlindungan Data Peribadi 2010)

Cara untuk protect e-masjid.my ?

**Comment by AimanFariz** [comment]
Soalan:
1. Adakah ahli kariah yang daftarkan anak mereka sendiri?
2. Di mana data anak kariah disimpan? Adakah dalam database ahli kariah tetapi dengan special tag/label? Atau buat database asing untuk anak kariah?
3. Adakah ahli kariah boleh lihat data semua anak kariah? Atau hanya visible ke ajk masjid seperti pengerusi/bendahari? perlukan  authorization utk ini?

https://www.pdp.gov.my/jpdpv2/akta-709/akta-perlindungan-data-peribadi-2010/
ini link rujukan untuk Akta Perlindungan Data Peribadi 2010.

**Reply by rohaizanr** [reply]
Jawapan:

1. Ya (Boleh tengok contoh googleform dan masjidpro)
2. Db/table asing utk anak kariah.
3. Hanya visible utk AJK

**Comment by mfar1984** [comment]
Megenai Isu Akta Perlindungan Data Peribadi 2010, sebagai yang bekerja dengan Law Firm, 

kita takda masalah untuk dapatkan data tersebut. sebab saya kalau tengok, data peribadi ni hanya minta nama penuh, ic, no tel, gambar,

dan bukan Salinan Kad Pengenalan. yang perlu dipatuh adalah 7 prinsip

7 PRINSIP PERLINDUNGAN DATA PERIBADI

Wajar orang ramai mengetahui bagaimana data peribadi kita diperolehi dan menyedari akan 7 Prinsip Perlindungan Data Peribadi yang disaran oleh APDP 2010 (Akta 709) iaitu :-
7 Prinsip Perlindungan Data Peribadi yang wajib dipatuhi di bawah seksyen 5(1) dalam Akta  709 ini demi menjaga keutuhan data peribadi :-
Pertama –  Prinsip Am di mana seseorang pengguna tidak dibenarkan memproses data peribadi seseorang lain tanpa kebenarannya. Pengertian proses di sini harus dimengertikan sebagai mengendalikan data melalui cara atau kaedah automatis atau pengkomputeran atau apa-apa proses lain.
Kedua – mesti mematuhi Prinsip Notis dan Pilihan di mana makluman dan tujuan awalan dimaklumkan kepada subjek data berkenaan.
Ketiga – ialah Prinsip Penzahiran tujuan data peribadi seseorang subjek itu demi mengenal pasti maksud yang baginya data peribadi itu hendak dizahirkan.
Keempat –  Prinsip Keselamatan – apabila memproses data peribadi mana-mana subjek, mengambil langkah supaya data tersebut selamat, tidak diubahsuai, disalahguna atau diberikan kepada pihak-pihak yang tidak berkenaan.
Kelima  – Prinsip Penyimpanan: sesuatu data peribadi itu tidak dibenarkan disimpan di dalam sesuatu pemprosesan lebih daripada had masa yang diperlukan.
Keenam  –  Prinsip Integriti Data – setiap data peribadi dipastikan supaya tepat, lengkap, tidak mengelirukan dan terkini menepati maksud sesuatu data itu disimpan dan diproses.
Ketujuh – Prinsip Akses: seseorang hendaklah diberi hak akses kepada data peribadinya yang dipegang oleh seseorang pengguna data dan juga boleh membetulkan datanya itu supaya terkini.
Oleh  yang demikian, rakyat Malaysia khususnya perlu menyedari hak-hak mereka mengikut prinsip-prinsip yang terdapat di dalam Akta ini.

diharap ini dapat menjawab soalan pihak tuan,

saya sarankan sesiapa yang ingin mengisi data tersebut, kita kena buat checkbox menyatakan Saya bersetuju untuk memberi maklumat untuk kegunaan Masjid xxxxxx

dan kalau masih risau untuk bab ni, saya sarankan module ini hanya boleh di access melalui intranet VPN > REVERSE PROXY > MODULE

**Comment by AimanFariz** [comment]
baik. 
![image](https://github.com/Dev4w4n/e-masjid.my/assets/99475959/07387a1d-5940-4a54-979a-d6da524bb729)
kiranya view 'Daftar' AJK Masjid adalah utk daftar ahli kariah, dan view 'Daftar' untuk ahli kariah adalah untuk daftar anak mereka?

**Reply by rohaizanr** [reply]
Tak perlu daftar di bahagian dashboard, sebab untuk fungsi daftar utk ahli kariah di public kan.

Apa yg perlu mungkin, Edit.

**Reply by rohaizanr** [reply]
Edit mungkin tak boleh kalau nak ikut Akta, 

Mungkin saya salah faham

@mfar1984 comments?

**Reply by AimanFariz** [reply]
ok saya faham.
![image](https://github.com/Dev4w4n/e-masjid.my/assets/99475959/094200a4-d33b-41a1-9da5-27d2fd503741)
ahli kariah pergi ke web public masjid -> E-Khairat, daftar maklumat tanggungan di situ?

**Comment by mfar1984** [comment]
Edit - boleh edit tapi kena ada form untuk ahli qariah tu edit, contohnya masa submit salah ejaan nama, jadi dia akan pegi ke masjid untuk minta pengurusan supaya memperbetulkan nama dia dalam tu atau seumpamanya, 

so dia boleh sign 1 surat yang dia nak ubah nama

kedua mengenai pendaftaran, kalau ahli di masjid tu terlalu banyak, pengurusan xmampu nak handle semua, rasanya boleh di public kan, tp dengan syarat supaya ahli tu di approve setelah melihat data yang dia bagi.

kalau ikut module, memang kena ada frontend dan backend untuk module kariah ni. tapi anak kariah tu kalau kita nak bagi pada ahli kariah tengok anak dibawah dia, maksudnya sistem ni akan jadi sistem online dan pendaftaran online.

lihat balik e-masjid ni tujuan dia sampai ke situ tak

**Comment by rohaizanr** [comment]
@AimanFariz 

Jaga-jaga jgn confuse khairat dgn kariah. Dua benda berbeza

**Reply by AimanFariz** [reply]
maaf sy tersilap tgk ejaan

**Comment by AimanFariz** [comment]
Ok, saya baru faham makna sebenar anak kariah dan khairat. Maaf atas kesilapan terma di atas. 

Berdasarkan carian setakat ini, Masjid Sungai Kechil punya pendaftaran ahli kariah bermula di sini:
https://www.buletinmutiara.com/wujudkan-pengkalan-data-anak-kariah-masjid-tkm-i/

Jadi soalan saya sekarang ini, memandangkan anak kariah akan mendaftarkan maklumat mereka sendiri, adakah boleh buat 2 cara pendaftaran:
1. kod QR - sediakan kod QR yang boleh diprint dan ditampal di masjid (physical), link ke portal kariah e-masjid untuk domain masjid tersebut. dekat situ, mungkin boleh ada butang 'Daftar Anak Kariah' dan 'Kemaskini data anak kariah'. kalau public klik kemaskini, mereka akan dibawa ke window utk masukkan beberapa detail (identification) mcm nama atau nombor ic, untuk tgk nama mereka dh ada belum dlm tu. klu match, mereka boleh update data di window seterusnya.
![image](https://github.com/Dev4w4n/e-masjid.my/assets/99475959/274359d8-a535-43d0-a0be-f1e85336d4bb)

2.  di website e-masjid, domain masjid tersebut, ada portal ahli kariah. boleh daftar dan update terus di situ.

Rasional:
1. penggunaan kod QR semakin berleluasa dan boleh memberikan user experience yang bagus serta pendaftaran yang cepat berbanding dengan public perlu ke website e-masjid utk masjid tersebut.
2. kita perlu mengambil kira faktor device public jugak. Mungkin public ada masalah untuk akses QR code dengan device, jadi mereka boleh pilih option kedua, iaitu layari laman web masjid.

**Reply by mfar1984** [reply]
Rasional:

1. penggunaan kod QR semakin berleluasa dan boleh memberikan user experience yang bagus serta pendaftaran yang cepat berbanding dengan public perlu ke website e-masjid utk masjid tersebut. **(SETUJU)**
2. kita perlu mengambil kira faktor device public jugak. Mungkin public ada masalah untuk akses QR code dengan device, jadi mereka boleh pilih option kedua, iaitu layari laman web masjid. **(SETUJU)**

Kena buat Documentation yang boleh auto generate bahagian pendaftaran atau pun masukkan dalam idea untuk writing Documentation

untuk pendaftaran anak kariah rasanya xperlu ada 1 module dibawah anak kariah, hanya siapa yang ahli kariah boleh access ke laman web dan dia ada access untuk daftar terus anak dibawah dia.

jadi pendaftaran anak kariah ni dibawah module ahli kariah, 

tujuan dia untuk kita senang nak match ibu bapa yang tersebut anak dia berapa orang, dan akan papar senarai nama anak dibawah mereka, andaikata kemaskini dibawah ahli kariah itu, itu saya setuju, tujuan dia lagi senang untuk kita verify data ni anak kariah dibawah jagaan siapa.

ini serupa dengan JPN sistem jgk.

**Reply by AimanFariz** [reply]
ok skrg ni saya keliru balik, adakah anak kariah ni sebenarnya merujuk kepada anak kepada ahli kariah?

Atau merujuk kepada ahli kariah/ahli mukim tu sendiri

**Reply by mfar1984** [reply]
yang ni anak kariah kita bincang, anak kepada ahli kariah tu sendiri kalau tak silap. @rohaizanr  mungkin boleh explain lagi, atau ni orang dibawah jagaan diaorang seperti ahli-ahli asnaf ? 

sbb bantuan asnaf ni akan diketahui oleh ahli kariah tu sendiri. apa pendapat tuan ?

**Comment by mfar1984** [comment]
> ok skrg ni saya keliru balik, adakah anak kariah ni sebenarnya merujuk kepada anak kepada ahli kariah?
> 
> Atau merujuk kepada ahli kariah/ahli mukim tu sendiri

terus saya blur jgk

ahli kariah tidak berdaftar - yang bermukim dan duduk disekitar masjid / kampung, lebih kurang dalam 4 bulan, yg selalu ikut program2 masjid / surau. yg boleh daftar semua orang dan tidak termasuk bukan warganegara dan penduduk luar kawasan tersebut.

ahli kariah berdaftar - perbezaan ahli kariah yang berdaftar, diaorang boleh ikut MAT, layak undi untuk MAK, layak jgk di calonkan di dalam mesyuarat agong khas dan lain2 lagi, keahlian diaorang ni terlucut andaikata dia meninggalkan mukim tersebut atau meninggal dunia

jadi untuk menjawab tuan nye kat atas, tuan nak buat ahli kariah jadi ahli kariah yang berdaftar kan ?

**Comment by rohaizanr** [comment]
@AimanFariz 

Ahli kariah = anak kariah
Sama je dua ni.
Panggilan anak kariah bunyinya lebih mesra dan dekat dengan masjid 😄

Tanggungan anak kariah = anak/isteri atau orang tua kepada ahli kariah tersebut. Kita tak sebut "anak kepada anak kariah", tanggungan lebih sesuai.

**Reply by AimanFariz** [reply]
Baik, terima kasih atas penerangan tuan

**Comment by mfar1984** [comment]
Akan ada tak module tambahan untuk anak kariah dimana nanti dia boleh masukkan tanggungan dibawah dia? Atau sekalikan sahaja di dalam anak kariah module?

**Reply by rohaizanr** [reply]
Kita boleh sekalikan saja.

**Reply by AimanFariz** [reply]
baik, kalau mcm tu sy boleh start design wireframe and Ui mockup?

**Comment by rohaizanr** [comment]
Pendaftaran Ahli Kariah (public)

Cadangan saya:
Kita buat mcm Step-by-step Wizard like form

Step 1:
Accept terma dan syarat.
Contoh: "Saya setuju untuk memberikan maklumat peribadi saya kepada MASJID untuk tujuan semakan dan rekod masjid"
Button "Setuju dan Teruskan"

Step 2:
Public data entry.
Isikan butiran Seperti dalam google form/masjidpro) diatas. Kita buat cantik lagi 😄

Step 3:
Preview and send.
Print out semua butiran dalam bentuk form yang senang dilihat. Pengguna boleh verify apa yang dia key in.
Di bawah sekali ada checkbox
✅ Saya seperti nama dan butiran di atas, memohon untuk didaftarkan sebagai Ahli Kariah bagi Kariah untuk MASJID ???
✅ Saya memperakui maklumat di atas adalah benar dan sekiranya didapati palsu maka permohonan ini adalah terbatal dengan sendirinya.

Button "Hantar"


Step 4
Success page.
Maklumat anda telah dihantar ke pihak masjid.

Button "Kembali ke laman utama"

(Data semua akan di email ke user, as reference)

**Reply by lurhays** [reply]
Macam file income tax tu ke?

**Reply by rohaizanr** [reply]
ya

**Comment by AimanFariz** [comment]
Assalamualaikum, ini cadangan UI Mockup saya.

https://github.com/Dev4w4n/e-masjid.my/assets/99475959/5382463a-47c7-4cf2-b3f4-e34ee0b46e6c
Cadangan ini berdasarkan cadangan Step-by-step wizard form oleh tuan @rohaizanr.

Box2 maklumat tu pula saya ambil daripada Google Form Masjid Sungai Kechil.

Saya cadangkan kita bahagikan maklumat2 tu kepada beberapa kategori:

Maklumat Peribadi:
Nama penuh
No. kad pengenalan
Tarikh lahir
Umur
Jantina

Kontak:
No. telefon
Alamat rumah
Muat naik gambar (boleh juga dikelompokkan dalam kategori lain bergantung pada konteks penggunaannya)

Maklumat Demografik:
Bangsa
Warganegara
Status perkahwinan
Warga emas atau tidak

Maklumat Social:
Jenis bantuan sedang diterima
Ada khairat kematian atau tidak
OKU atau tidak
Wajib solat Jumaat atau tidak

Maklumat Pekerjaan:
Pekerjaan - ada butang utk tambah, andainya public kerja lebih dari satu pekerjaan

Keluarga/Tanggungan:
Maklumat tanggungan - pun ada butang untuk tambah

Pengesahan:
Checkbox "Saya berjanji bahawa maklumat yang diberikan adalah betul" lebih kurg mcm ni,

Lepas click checkbox, boleh click "Hantar"

Maklumat Tambahan: Terma dan Syarat tu dalam cadangan di atas, dia adalah link, so kalau public klik, akan direct ke page/popup Terma dan Syarat

Sila review cadangan ini dan berikan feedback untuk penambahbaikan

**Reply by rohaizanr** [reply]
@AimanFariz ada sikit. perubahan.

step1: Accept terma dan syarat.
step2: masuk no IC (backend buat search)
step3: sama dan seterusnya

**Reply by rohaizanr** [reply]
Wajib solat Jumaat atau tidak (boleh buang)

**Comment by mfar1984** [comment]
cantik dah tu, step by step wizard ni rasanya kita cuma buat 3 sahaja

1. terma dan syarat untuk diaorang accept
2. form ( buat box in box - nampak lebih kemas )
3. review ( ada button download.& print pdf

<img width="970" alt="Screenshot 2024-03-20 at 14 04 46" src="https://github.com/Dev4w4n/e-masjid.my/assets/33825287/b9bb3535-bedb-4cc8-a7e1-c8192ac20f24">

macam ni pun cantik box in box kalau nak buat

**Comment by rohaizanr** [comment]
@AimanFariz ada idea untuk dashboard ?

**Reply by AimanFariz** [reply]
apa perbezaan requirement dekat Dashboard dan Public ya? adakah ajk boleh edit maklumat di dashboard? atau hanya boleh search ahli? kalau hanya boleh search, boleh search ikut nama dan pakai tag mcm yg sedia ada utk khairat ke?

**Reply by rohaizanr** [reply]
kita buat AJK tak boleh edit, ikut AKTA PDPA.

AJK boleh list all , search by name/ic dulu.

buat minimum feature, tengok requester nanti nak mintak apa, kita tambah

**Comment by mfar1984** [comment]
User / AJK yang dilantik dan ada akaun sendiri xboleh EDIT file tu melainkan Admin sahaja.

balik pada soalan sebelum ni saya pernah tanye jgk, xde masalah ye kita buat kemudian user level tu?

**Reply by rohaizanr** [reply]
sekarang ni memang dah ada Admin user iaitu saya(superuser).

**Comment by mfar1984** [comment]
Ada penambahan sikit @AimanFariz , mengenai form step by step wizard itu dah terbaik, tp kalau dapat, apabila ada user nak daftar dan sekiranya failed to submit seperti verification ROBOT ke ape, or failed sbb xde lain. Harap ada form cache, diaorang xperlu lg isi data yang sama.

**Reply by rohaizanr** [reply]
Insyallah, ini kita akan buat

**Comment by mfar1984** [comment]
Untuk comply supaya penyalahgunaan data tidak disalahguna, saya sarankan kita buat seperti data PRU dimana pihak yang ingin tahu status ahli tersebut harus masukkan no ic sahaja.

kalau search menggunakan nama, data mudah untuk dicari apatah lagi kalau hanya letak 1 huruf tapi data dah boleh dapat dicari.

jadi ini berkaitan dengan 7 prinsip pdpa.

dan dalam kes di atas, untuk prinsip ke-7,

pihak ahli khairat atau kariah wajib mendaftar di laman web tersebut supaya mereka boleh access data mereka sendiri.

ini juga salah 1 di dalam 7 prinsip akta pdpa yang harus di patuhi.

tujuan adalah mereka boleh memperbetulkan data peribadi mereka sendiri. Tetapi approval daripada pihak admin itu sendiri

sekaligus menjawab soalan apakah kita perlu user level access, jawapan adalah YA

**Reply by rohaizanr** [reply]
Saya nak simplify flow ni.

Data akan di email ke ahli kariah setelah register.

Kalau ada perubahan, dia perlu reply email tu ke superadmin.

**Reply by mfar1984** [reply]
Kalau yang ubah data tu dari pihak admin, maksudnya dia x comply dalam 7 prinsip akta pdpa.

**Reply by rohaizanr** [reply]
Ok faham.

Jadi di skrin pendaftaran tu, setelah user setuju, perlu isikan no IC dulu.

Ini untuk backend tarik semua data yg link dengan IC tu, jika ada, user boleh edit. Jika tidak, user akan isikan maklumat baru.

Step 1: Agreement page
Step 2: Search No IC
Step 3-seterusnya: Isi maklumat

Boleh macam ni @mfar1984 ?

**Comment by mfar1984** [comment]
Bukan, untuk case carian IC by ADMIN atau Ahli Kariah melalui backend dashboard atau admin dashboard. 

perlu masukkan ic sahaja dan data tersebut akan ditarik sekiranya data tersebut ada dibawah ic tersebut.

untuk admin sahaja boleh edit, untuk ahli kariah atau ajk hanya boleh view only.

tapi untuk pendaftaran pulak, setiap user harus daftar dan pendaftaran tersebut mestilah approval oleh pihak ajk secara manual.

Jadi user yang daftar tersebut boleh membuat edit user profile dgn tempat tertentu tetapi tetap juga pihak ADMIN akan membuat verifikasi dan kelulusan pendaftaran atau ubahsuai data tersebut dr pihak user.

**Reply by Dev4w4n** [reply]
now i'm confused.

> Kalau yang ubah data tu dari pihak admin, maksudnya dia x comply dalam 7 prinsip akta pdpa.

atau

> untuk admin sahaja boleh edit, untuk ahli kariah atau ajk hanya boleh view only.

yang mana satu nak ikut ni

**Reply by mfar1984** [reply]
Mcm ni bro,

Edit utk profile user adalah user sendiri,

kita boleh approve untuk ubahsuai dari user itu sendiri.

Kat sini dh comply utk akta pdpa.

Admin boleh edit, kalau nak letak admin boleh edit. Tp jalan terbaik adalah, kena user mesti boleh edit jgk.

edit tu bahagian tertentu, seperti no tel, alamat n lain. Untuk bahagian ic nama, email dan lain2 mungkin boleh dibincangkan.

sbbnya masa approval tu, ajk masjid atau admin pasni dah verify ic dan alamat mesti sama seperti dalam pendaftaran dan ic user. 

Syarat kita dh penuhi dimana user boleh edit, itu je yg penting, kalau admin nk buat edit button pun xde masalah, kalau xbuat pun xde masalah. Tp user mesti boleh edit profile dia.

**Reply by rohaizanr** [reply]
Ok macam ni.

<hr />

**Dashboard AJK: (Modul baru Ahli Kariah)**
1. Admin boleh search profile user tu.
2. Open user profile
3. Admin boleh click untuk _Allow user edit profile_ button (approval) 

<hr />

**Public-web: (Pendaftaran Anak Kariah page)**
**Step 1:**
Accept terma dan syarat.
Contoh: "Saya setuju untuk memberikan maklumat peribadi saya kepada MASJID untuk tujuan semakan dan rekod masjid"
Button "Setuju dan Teruskan"

**Step 2:**
Masukkan No IC. Teruskan.


**Step 3:**
(Tarik maklumat penuh yang link dengan No IC)

(Jika ada)
- Boleh view sahaja (by default)
- Boleh edit (sekinyanya, admin click _Allow user edit profile_ button di Dashboard)

(Jika tiada maklumat berkaitan no IC, isi maklumat baru)
 - Public data entry.

**Step 4:**
Preview and send.
Print out semua butiran dalam bentuk form yang senang dilihat. Pengguna boleh verify apa yang dia key in.
Di bawah sekali ada checkbox
✅ Saya seperti nama dan butiran di atas, memohon untuk didaftarkan sebagai Ahli Kariah bagi Kariah untuk MASJID
✅ Saya memperakui maklumat di atas adalah benar dan sekiranya didapati palsu maka permohonan ini adalah terbatal dengan sendirinya.

<hr />


Jadi senario ini boleh handle semua kes dan comply dengan Akta PDPA.
Kalau ahli kariah nak edit profile, perlu inform AJK masjid untuk _buka lock_ profile tersebut.

**Reply by mfar1984** [reply]
Tp step 1 sepatutnya kena login dulu user, kalau pendaftaran baru terus pendaftaran baru, dan user tersebut lah boleh edit profile tu. 

Xperlu check ic, tp kalau dia register lagi kali kedua tp ada persamaaan seperti no ic, no telefon dan email. Popup akan keluar bgtahu user exist. Itu je rasanya.

**Reply by rohaizanr** [reply]
Tak perlu complicated kan feature mudah ni dgn user login.

Boleh try search dan baca MVP, minimal viable product.

Contoh di post pertama Masjid Pro TIADA LOGIN dan patuh PDPA.

https://www.masjidpro.com/Masjid/daftar_online/pendaftaran.php?redirect=&id_masjid=3744

Boleh proceed dah ni. Saya dah boleh create new issue.

Thanks @mfar1984

**Reply by mfar1984** [reply]
Boleh jgk buat mcm tu.

**Reply by AimanFariz** [reply]
Assalamualaikum,

Sorry lama x engage in discussion. Ok skrg sy faham the workflow, tuan nk sy buat UI mockup baru? Utk yg 4 step kt atas tu. Kalau perlu, sy boleh buatkan utk 2 kes mcm yg kt atas tu la. Kes yg maklumat dh ada, and kes yg maklumat belum ada, and sy boleh include captcha box mcm yg dicadangkan jugak.

**Reply by rohaizanr** [reply]
Boleh jugak Aiman. Terima kasih

**Comment by AimanFariz** [comment]
Assalamualaikum. Ini cadangan UI berdasasrkan perbincangan terkini.

Sections:
1. Perakuan
2. No. Pengenalan
3. Maklumat Diri
4. Pengesahan

Perakuan:
Sebelum tick:
![image](https://github.com/Dev4w4n/e-masjid.my/assets/99475959/51bb906e-abaa-462f-a866-4fdb4899775b)
Selepas tick:
![image](https://github.com/Dev4w4n/e-masjid.my/assets/99475959/6b107f27-e0ce-4f76-9834-44a0174c9cf9)

No. Pengenalan:
![image](https://github.com/Dev4w4n/e-masjid.my/assets/99475959/86a73984-2303-4b34-b156-38ba127bd63b)
Jenis Pengenalan: No. Passport, MyKad, etc. (yg bersesuaian)

Maklumat Diri:
![image](https://github.com/Dev4w4n/e-masjid.my/assets/99475959/cde467e3-02c6-4485-a9cc-e358e891b708)
![image](https://github.com/Dev4w4n/e-masjid.my/assets/99475959/3a2e06a7-1a70-4797-a00d-a2e59b830c34)
![image](https://github.com/Dev4w4n/e-masjid.my/assets/99475959/15852942-d028-4755-b480-0eafce36d41a)
![image](https://github.com/Dev4w4n/e-masjid.my/assets/99475959/ffc19bd2-997d-4b41-ad12-f0c02c26a88f)

Penambahan: Box-in-box
Warna box-in-box:
- Default : Putih
- Current: Gradient biru
- Complete: Hijau

Lepas semua dh hijau, boleh click 'Teruskan' dekat box terakhir (Keluarga)
Pengesahan:
![image](https://github.com/Dev4w4n/e-masjid.my/assets/99475959/f35947c1-2240-419e-98f0-c128ffcbdfd2)
![image](https://github.com/Dev4w4n/e-masjid.my/assets/99475959/47d3550b-3468-4ead-b94e-83ca9a820ff3)

Selepas hantar:
![image](https://github.com/Dev4w4n/e-masjid.my/assets/99475959/ebf859b4-815e-4685-a716-51ad63382ae3)

**Reply by rohaizanr** [reply]
@AimanFariz cantik dah ni. 

kalau dekat "Maklumat diri" , tambah button "Teruskan", make sense tak ?

**Reply by AimanFariz** [reply]
untuk setiap sub section ke? Saya rasa make sense je, sbb mungkin ada user yg x familiar dgn box-in-box, lepas dh isi kt bahagain Peribadi dia xtau nk proceed ke next page mcm mana

**Comment by rohaizanr** [comment]
updated project plan https://github.com/users/Dev4w4n/projects/1

@AimanFariz untuk dashboard dah ada idea ?

**Reply by AimanFariz** [reply]
Sy dh lost in the comments actually pasal dashboard ni.

Ada berapa jenis dashboard dan apa requirements yg perlu ada? Dashboard utk siapa dan apa deature yg user ni boleh buat di diashboard?

**Reply by rohaizanr** [reply]
Dashboard untuk kegunaan AJK masjid.

Yang UI kat atas tu untuk public-web, anak kariah isi.

Akan ada side menu baru di Dashboard:

Ahli Kariah
- Carian (by nama, IC, yang berpenyakit, yang menerima bantuan, yang OKU, etc;)
- - View Profile (Modal popup?) - Di sini akan ada toggle untuk benarkan anak kariah edit profile di public-web
- - Boleh print senarai (yang berpenyakit, yang menerima bantuan, yang OKU, etc;)
- QR (boleh print macam cadangan)

**Reply by rohaizanr** [reply]
Hmmm... Rasanya straight forward design kot, tak perlu Figma.

**Reply by AimanFariz** [reply]
Ok faham. saya fikir design dulu
